docs/guides/nwaku/build-source.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Build Nwaku from Source
+---
+
+This guide provides detailed steps to build a `nwaku` node from the source to access the latest development version or a specific commit or tag of `nwaku`. If you prefer a more stable version, [download a pre-compiled binary](https://github.com/waku-org/nwaku/tags) instead.
+
+:::info
+- A minimum of 2GB of RAM is required to build `nwaku`.
+- Nwaku is available for Linux and macOS, with experimental Windows support.
+:::
+
+## Install Dependencies
+
+To build nwaku, you will need the standard developer tools, including a C compiler, Make, Bash, and Git.
+
+#### Linux
+
+To install the dependencies on common Linux distributions, use:
+
+```mdx-code-block
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+```
+
+<Tabs>
+<TabItem value="debian" label="Debian and Ubuntu">
+
+```bash
+sudo apt-get install build-essential git
+```
+
+</TabItem>
+<TabItem value="fedora" label="Fedora">
+
+```bash
+dnf install @development-tools
+```
+
+</TabItem>
+<TabItem value="arch" label="Arch Linux">
+
+```bash
+# Using your favorite AUR helper
+[AUR HELPER] -S base-devel
+```
+
+</TabItem>
+</Tabs>
+
+#### macOS
+
+If you use [Homebrew](https://brew.sh/) to manage packages, use:
+
+```bash
+brew install cmake
+```
+
+## Clone the Repository
+
+Get the source code from the GitHub repository. The default branch is `master`, the release candidate for major updates.
+
+```bash
+git clone https://github.com/waku-org/nwaku
+cd nwaku
+```
+
+:::tip
+You can use `git tag -l` to check specific version tags.
+:::
+
+## Build the Binary
+
+Build the `nwaku` binary:
+
+```bash
+make wakunode2
+```
+
+The first `make` invocation updates to all Git submodules. After each `git pull`, run `make update` to keep the submodules updated in the future.
+
+```bash
+make update wakunode2
```

Comment:
```suggestion
make update
```

I don't think `wakunode2` changes anything. @jm-clius to confirm.

Reply:
Both of them work fine, I'll just use `make update` in the docs